src/backup.rs

```diff
@@ -26,17 +29,26 @@ struct Backup {
 
 
 impl BackupOptions {
+    pub fn with_excludes(excludes: Option<Vec<&str>>) -> Result<Self> {
+        let mut backup_options = BackupOptions::default();
```

Comment:
This should consume a BackupOptions and return a new one, so the call looks like

    BackupOptions::default().with_excludes(...).backup(...)